src/zui/ZUIButton/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button, CircularProgress } from '@mui/material';
+import { FC, KeyboardEventHandler, MouseEventHandler } from 'react';
+
+type ZUIButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'tertiary'
+  | 'destructive'
+  | 'warning'
+  | 'loading';
+
+interface ZUIButtonProps {
+  buttonType?: 'button' | 'reset' | 'submit';
+  disabled?: boolean;
+  endIcon?: JSX.Element;
+  fullWidth?: boolean;
+  label: string | JSX.Element;
```

Comment:
Do we have to allow `JSX.Element` here? I feel like that encourages breaking the mold. The one scenario I can imagine where allowing an element makes sense is when we have `<Msg>` as children. But I'm thinking maybe we could solve that by passing a message ID as the label instead?

Reply:
i think i made it like this for when the button only contains an icon, like the ellipsis in the ellipsis menu. It can probably be solved differently!